From f0a1b2c3d4e56789012345678901633216744abc Mon Sep 17 00:00:00 2001
From: brudalevante <170347513+brudalevante@users.noreply.github.com>
Date: Mon, 15 Sep 2025 19:40:31 +0000
Subject: [PATCH] net: phy: sfp: add quirk for H-COM SPP425H-GAB4 XGS-PON module

Add quirk for the H-COM SPP425H-GAB4 XGS-PON module to use sfp_fixup_halny_gsfp.
This module requires the fixup for correct operation.

The module identifiers are based on working EEPROM and ethtool output.

Signed-off-by: OpenWrt <openwrt@openwrt.org>
---
drivers/net/phy/sfp.c | 4 ++++
1 file changed, 4 insertions(+)

diff --git a/drivers/net/phy/sfp.c b/drivers/net/phy/sfp.c
index 14f14bc..7b2a6a1 100644
--- a/drivers/net/phy/sfp.c
+++ b/drivers/net/phy/sfp.c
@@ -513,6 +513,10 @@
 	
 	// OEM XGSPONST2000 XGS-PON module (SFP+)
 	SFP_QUIRK_F("OEM", "XGSPONST2000", sfp_fixup_halny_gsfp),
+	
+	
+	// OEM XGSPONST2001 XGS-PON module (SFP+)
+	SFP_QUIRK_F("OEM", "XGSPONST2001", sfp_fixup_halny_gsfp),	
 
 	// HG MXPD-483II-F 2.5G supports 2500Base-X, but incorrectly reports
 	// 2600MBd in their EERPOM
-- 
2.45.2

